framework/src/main/java/io/github/kgress/scaffold/BaseComponent.java

```diff
@@ -0,0 +1,234 @@
+             parent locator.
+             */
+            var elementBy = listOfElements.get(index).getBy();
+
+            /*
+            Check to see if the parentBy actually exists in this case. If there are any elements
+            that have been constructed with a parent in mind, we want to make sure this parent
+            isn't a xpath locator, along with the current elementBy.
+             */
+            var elementParentBy = listOfElements.get(index).getParentBy();
+
+            /*
+             Check to make sure the By locator for the parent is a type of CSS selector, where
+             type is anything other than XPATH. Then, get the underlying locator as string.
+             */
+            if (elementBy instanceof By.ByXPath || elementParentBy instanceof By.ByXPath) {
+              throw new ComponentException("Scaffold currently cannot build component lists using "
+                  + "XPATH. Please use By locators that are a type of Css selector.");
+            }
+            var underlyingSelector = AutomationUtils.getUnderlyingLocatorByString(elementBy);
+
+            /*
+             Create a new locator that combines the parent (the underlyingLocator) and
+             an :nth-child using the index. Because the element list starts at 0, we
+             need to add 1 in order to adhere to correct CSS usage.
+             */
+            var fullNewSelector = String.format("%s:nth-child(%s)", underlyingSelector,
+                index + 1);
+
+            /*
+             Create a new instance of the component passed in by the caller. The class
+             extending off of BaseComponent should not have a constructor, otherwise this new
+             instance will fail.
+             */
+            var componentInstance = component.getConstructor().newInstance();
+
+            /*
+             Iterate through the list of fields on the new instance of the component.
+             We should only convert strong typed Scaffold elements but allow for
+             additional fields, such as Strings (e.g. if Strings are being used as
+             locators).
+             */
+            convertFieldsWithNewLocator(componentInstance, fullNewSelector);
+
+            /*
+             After the fields have been converted on the new instance of the component,
+             add it to the list that we will return to the caller.
+             */
+            listOfComponents.add(componentInstance);
+          } catch (InstantiationException | IllegalAccessException | InvocationTargetException |
+              NoSuchMethodException e) {
+            throw new ComponentException(e);
+          }
+        });
+    return listOfComponents;
+  }
+
+  /**
+   * Converts {@link Field}'s from a class that extends off of {@link BaseComponent} from an
+   * "inaccessible" state to "accessible." We will only modify the access of Scaffold strongly typed
```

Comment:
* We will only convert Scaffold elements. Access is still technically modified for every field, but always set back to private.